### ISSUE TITLE
Something unwraps "Too many open files" errors

### ISSUE THREAD

**Issue by skade** [http-rs/surf#29]
# Bug Report
## Your Environment
| Software         | Version(s) |
| ---------------- | ---------- |
| surf      | master
| Rustc            | rustc 1.38.0-nightly (534b42394 2019-08-09)
| Operating System | Linux
| Client Type | chttp

## Expected Behavior
All connection failures should be reported to the user to handle.

## Current Behavior

```
thread 'async-task-driver' panicked at 'client failed to initialize: Io(Os { code: 24, kind: Other, message: "Too many open files" })', src/libcore/result.rs:1084:5
```

## Code Sample

https://github.com/async-std/async-crawler/blob/master/src/main.rs#L52

Set crawl depth to `3` on a machine with a 1024 open files limit.

**Comment by yoshuawuyts** [comment]
Related issue: https://github.com/sagebind/isahc/issues/62

**Comment by sagebind** [comment]
https://github.com/sagebind/isahc/issues/62 has now been fixed in Isahc v0.7.0; upgrading should fix this issue.
